### PR TITLE
IGVF-890-report

### PR DESCRIPTION
**Pull request by mingjiecn**

### CODE REVIEW

File: src/igvfd/search_views.py [hunk 1]
```diff
@@ -117,7 +117,7 @@ def report(context, request):
             ),
             IDResponseField(),
             ContextResponseField(),
-            BasicReportWithFacetsResponseField(
+            MultipleTypesReportWithFacetsResponseField(
```
Comment (keenangraham) [inline src/igvfd/search_views.py:120]: Make a new report endpoint instead of changing this one. Something like `multitype-report` or `multireport`?

File: src/igvfd/report.py [hunk 2]
```diff
@@ -56,20 +62,37 @@ def _convert_camel_to_snake(type_str):
...
+        },
+        response_fields=[
+            MultipleTypesReportWithFacetsResponseField(
+                default_item_types=DEFAULT_ITEM_TYPES,
+                reserved_keys=RESERVED_KEYS,
+            ),
+            ColumnsResponseField(),
+
+        ]
+    )
```
Comment (keenangraham) [inline src/igvfd/report.py:78]: We should be calling the existing view instead of recreating it here. Something like `response = request.embed(f'/multitype-report/{original_query_string_with_limit_0_added}')`

File: src/igvfd/report.py [hunk 3]
```diff
@@ -56,20 +62,37 @@ def _convert_camel_to_snake(type_str):
...
+    facets = response.json['facets']
+    abstract_types = get_abstract_types(request)
+    types_in_search_result = []
+    for facet in facets:
+        if facet['field'] == 'type':
+            for term in facet['terms']:
+                type_name = term['key']
+                if type_name not in abstract_types:
+                    types_in_search_result.append(term['key'])
+            break
```
Comment (keenangraham) [inline src/igvfd/report.py:89]: What is the `break` doing?

File: src/igvfd/tests/indexing/test_report_tsv.py [hunk 4]
```diff
@@ -26,6 +26,26 @@ def test_batch_download_report_download(workbook, testapp):
         b'Contact P.I.', b'Title'
     ]
 
+    res = testapp.get('/report.tsv?type=Award&field=contact_pi&field=title&config=Award')
```
Comment (keenangraham) [inline src/igvfd/tests/indexing/test_report_tsv.py:29]: Make new tests specifically for `multitype-report`.

File: src/igvfd/report.py [hunk 5]
```diff
@@ -56,20 +62,37 @@ def _convert_camel_to_snake(type_str):
 @view_config(route_name='report_download', request_method='GET')
 def report_download(context, request):
```
Comment (keenangraham) [inline src/igvfd/report.py:63]: Leave this endpoint alone and make new endpoint e.g. `multitype_report_download`.

File: README.md [hunk 6]
```diff
@@ -56,6 +56,26 @@ $ pytest
 docker compose down -v
 ```
 
+Or run indexer tests interactively:
```
Comment (keenangraham) [inline README.md:59]: Would it be cleaner to put this in its own PR?

**Review by keenangraham — CHANGES_REQUESTED** [review]

File: src/igvfd/report.py (hunk 3 above)
Comment (mingjiecn) [reply src/igvfd/report.py:89]: break after find facet "type" and get all the term keys.

File: src/igvfd/search_views.py (hunk 1 above)
Comment (mingjiecn) [reply src/igvfd/search_views.py:120]: Sounds good, will do.

File: src/igvfd/report.py (hunk 2 above)
Comment (mingjiecn) [reply src/igvfd/report.py:78]: Got it. Thanks!
